### PR TITLE
Client/Server Integration for Profile Photo Upload

### LINKED ISSUE #41: Setup/IN: Amazon S3 or Cloudinary
Setup Amazon S3 or Cloudinary for profile picture uploading. Create the proper route for handling a user uploading their profile photo and integrate this in the front-end. 

If we're doing file attachments then please utilise Amazon S3.

### PR DESCRIPTION
**Pull request by EdwardOlmos**
### What this PR does (required):
- Resolves #41
- Modifies UI for photo upload and deletion
- Adjusts client interfaces to handle profile changes
- Added controller and route for delete functionality

### Screenshots / Videos (front-end only):
chrome_dAT47ZftqB

### Any information needed to test this feature (required):

1. Sign up as a new user and upload a profile photo from the navbar
2. It may take a second for the S3 link to be ready. Once uploaded, see the change.
3. Delete the profile photo, and see the change.

### CODE REVIEW

File: server/controllers/upload.js [hunk 1]
```diff
@@ -1,20 +1,68 @@
...
+          profile
+        }
+      });
+    }
+  }
+});
+
+// @route DELETE /uploads/profile
+// @desc Delete user's profile photo
+// @access Public
```
Comment (ezeilo-su) [inline server/controllers/upload.js:29]: Good documentation

**Review by ezeilo-su — COMMENTED** [review]
Hey @EdwardOlmos, good work with this ticket.

File: client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx [hunk 2]
```diff
@@ -16,12 +17,35 @@ interface ProfilePhotoProps {
...
+          updateProfileContext(data.success);
+          updateSnackBarMessage('Successfully uploaded file!');
+        } else {
+          updateSnackBarMessage('An unexpected error occurred. Please try again');
+        }
+      });
+    }
+  };
+
+  const onClick = () => {
```
Comment (rajivtitus) [inline client/src/pages/Settings/ProfilePhoto/ProfilePhoto.tsx:38]: Let's give this function a more descriptive name 😄

**Review by rajivtitus — APPROVED** [review]
Fabulous work! 💯 
Just a minor comment to address before merging 👍🏽

**Comment by EdwardOlmos** [comment]
Thank you, @rajivtitus! 👍

**Pull request merged**